app/waf/src/controllers/application.controller.ts

```diff
@@ -155,4 +173,79 @@ export class ApplicationController {
   async deleteById(@param.path.string('id') id: string): Promise<void> {
     await this.applicationRepository.deleteById(id);
   }
+
+  @post(prefix + '/applications/{id}/deploy', {
+    responses: {
+      '204': {
+        description: 'Application deploy success',
+      },
+    },
+  })
+  async deployById(@param.path.string('id') id: string): Promise<Object> {
+    let application = await this.applicationRepository.findById(id);
+
+    let params: {[key: string]: Object} = {
+      application: application,
+    };
+
+    let tenantAssoc = null;
+    try {
```

Comment:
`findById` is widely used, but it's not the best way. 
Why not use `find` with filter: 

```
    let tenantAssoc: TenantAssociation[];
    tenantAssoc = await this.tenantAssociationRepository.find({
      where: {
        tenantId: application.tenantId
      }
    });
    if (tenantAssoc.length === 0) throw new HttpErrors.UnprocessableEntity("xxxxx");
```

Reply:
Done